tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_component.ts

```diff
@@ -39,33 +41,44 @@ export interface ColorGroup {
 export class RegexEditDialogComponent {
   @Input() regexString!: string;
   @Input() colorRunPairList!: ColorGroup[];
+  @Input() selectedGroupBy!: GroupByKey;
 
-  @Output() onSave = new EventEmitter<string>();
+  @Output() onSave = new EventEmitter();
   @Output() regexInputOnChange = new EventEmitter<string>();
+  @Output() regexTypeOnChange = new EventEmitter<GroupByKey>();
 
   timeOutId = 0;
   @ViewChild('regexStringInput', {static: true})
   regexStringInput!: ElementRef<HTMLInputElement>;
+  regexMatchType = '';
 
   constructor(
     public readonly dialogRef: MatDialogRef<RegexEditDialogComponent>,
-    private readonly hostElRef: ElementRef
+    private readonly hostElRef: ElementRef,
+    private cdRef: ChangeDetectorRef
   ) {}
 
+  ngOnInit() {
+    this.regexMatchType =
```

Comment:
I think we can implement this enum->string logic a bit more concisely, especially because it's preferable to avoid two way data binding (`[(value)]`) in the template and also manual ChangeDetectorRef manipulation. I don't think we actually need a separate variable.

In this component file, defined a memoized helper function:
```
  regexTypeFn = memoize(this.internalRegexTypeFn);
  private internalRegexTypeFn(key: GroupByKey) {
    return key === GroupByKey.REGEX_BY_EXP ? 'regex_by_exp' : 'regex_by_run';
  }
```

In the template file:
```
      <mat-select
        [value]="regexTypeFn(selectedGroupBy)"
      ...
```

This will handle changes automatically on any selectedGroupBy (Input) changes. It needs to be memoized to be used in the template or it will be horribly inefficent.

Reply:
Oh didn't know about memoize. Will definitely implement this way!